hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/mode/ClientModeStrategy.java

```diff
@@ -0,0 +1,154 @@
+        Field.USER_COUNT);
+  }
+
+  List<Record> createRecords(ClusterMetrics clusterMetrics) {
+    List<Record> ret = new ArrayList<>();
+    for (ServerMetrics serverMetrics : clusterMetrics.getLiveServerMetrics().values()) {
+      long lastReportTimestamp = serverMetrics.getLastReportTimestamp();
+      serverMetrics.getUserMetrics().values().forEach(um -> um.getClientMetrics().values().forEach(
+          clientMetrics -> ret
+              .add(createRecord(um.getNameAsString(), clientMetrics, lastReportTimestamp))));
+    }
+    return ret;
+  }
+
+  /**
+   * Aggregate the records and count the unique values for the given distinctField
+   *
+   * @param records               records to be processed
+   * @param groupBy               Field on which group by needs to be done
+   * @param distinctField         Field whose unique values needs to be counted
+   * @param uniqueCountAssignedTo a target field to which the unique count is assigned to
+   * @return aggregated records
+   */
+  List<Record> aggregateRecordsAndAddDistinct(List<Record> records, Field groupBy,
+      Field distinctField, Field uniqueCountAssignedTo) {
+    List<Record> result = new ArrayList<>();
+    records.stream().collect(Collectors.groupingBy(r -> r.get(groupBy))).values()
+        .forEach(val -> {
+          Set<FieldValue> distinctValues = new HashSet<>();
+          Map<Field, FieldValue> map = new HashMap<>();
+          for (Record record : val) {
+            for (Map.Entry<Field, FieldValue> field : record.entrySet()) {
+              if (distinctField.equals(field.getKey())) {
+                //We will not be adding the field in the new record whose distinct count is required
+                distinctValues.add(record.get(distinctField));
+              } else {
+                if (field.getKey().getFieldValueType() == FieldValueType.STRING) {
+                  map.put(field.getKey(), field.getValue());
+                } else {
+                  if (map.get(field.getKey()) == null) {
+                    map.put(field.getKey(), field.getValue());
+                  } else {
+                    map.put(field.getKey(), map.get(field.getKey()).plus(field.getValue()));
+                  }
+                }
+              }
+            }
+          }
+          // Add unique count field
+          map.put(uniqueCountAssignedTo, uniqueCountAssignedTo.newValue(distinctValues.size()));
+          result.add(Record
+              .ofEntries(map.entrySet().stream().map(k -> Record.entry(k.getKey(), k.getValue()))));
+        });
+    return result;
+  }
+
+  Record createRecord(String user, UserMetrics.ClientMetrics clientMetrics,
+      long lastReportTimestamp) {
+    Record.Builder builder = Record.builder();
+    String client = clientMetrics.getHostName();
+    builder.put(Field.CLIENT, clientMetrics.getHostName());
+    RequestCountPerSecond requestCountPerSecond = requestCountPerSecondMap.get(client + "_" + user);
```

Comment:
It seems like we need to include a server name to the key of the requestCountPerSecondMap as the metrics are created per RS.